File: js/linearSearch.js
const arrayToSearchThrough = [];
for (let i = 1; i <= 1000; i++) {
    arrayToSearchThrough.push(i);
}

// Basic version
exports.linearSearch = function(valueToFind, arrayToSearchThrough) {
    for (let i = 0; i < arrayToSearchThrough.length; i++) {
        if (arrayToSearchThrough[i] === valueToFind) {
            return i;
        }
    }
    return undefined;
};

// Global version
exports.linearSearchGlobal = function(valueToFind, arrayToSearchThrough) {
    let indicesOfValues = [];
    for (let i = 0; i < arrayToSearchThrough.length; i++) {
        if (arrayToSearchThrough[i] === valueToFind) {
            indicesOfValues.push(i);
        }
    }

    return indicesOfValues;
};

File: python/linear_search.py
array_to_search_through = []
for number in range(1, 1001):
    array_to_search_through.append(number)


## linear search
def linear_search(value_to_find, array_to_search_through):
    index = None

    for i in range(len(array_to_search_through)):
        if (array_to_search_through[i] == value_to_find):
            index = i
    
    return index


## global search
def linear_search_global(value_to_find, array_to_search_through):
    indices_of_values = []

    if not array_to_search_through:
        return []

    for i in range(len(array_to_search_through)):
        if (array_to_search_through[i] == value_to_find):
            indices_of_values.append(i)

    return indices_of_values